src/System.Text.Json/src/System/Text/Json/Utf8JsonReader_TryGet.cs

```diff
@@ -25,7 +24,7 @@ public bool TryGetValueAsString(out string value)
             if (TokenType != JsonTokenType.String && TokenType != JsonTokenType.PropertyName)
                 return false;
 
-            byte[] valueArray = IsValueMultiSegment ? ValueSequence.ToArray() : ValueSpan.ToArray();
+            byte[] valueArray = ValueSpan.ToArray();
```

Comment:
netcoreapp21 added Span-based overloads to the Encoding classes, you should be able to avoid the redundant array.

The UWP compile might not see those, so you might need to \#if the code.

Reply:
You are looking at an out-dated change.